### PR TITLE
Window functions redux

### PR DESCRIPTION
**Pull request by arahlin**
Updates to the window function calculation for consistency with the paper text.

Cb / Db window functions are now linear combinations of qb window functions, weighted by a matrix of derivatives.

The qb window function calculation has also been adjusted to include the binning operator in the normalization instead of in the definition of M_ll'.

Finally, the handling of the `return_cls` and `cls_shape` variables has been cleaned up, so that they are now attributes that are passed around internally, rather than arguments that must be passed between methods.

The resulting bandpowers and error bars differ only slightly from the original implementation., and mainly at large scales, where the correlations between bins are more significant.

### CODE REVIEW

**Review by annegambrel — APPROVED** [review]
This looks great! It's all consistent with the text in the paper and I think we're good to merge, other than my question about that bin spectrum function-- want to double check you think it's useless now as well!

File: xfaster/xfaster_class.py [hunk 1]
```diff
@@ -4910,39 +4912,88 @@ def do_qb2cb(
...
+            Binned spectrum covariance
         qb2cb : dict
-            The conversion factor from ``qb`` to ``cb``, computed by
-            averaging over the input shape spectrum.
+            The conversion matrix from ``qb`` to ``cb`` for each spectrum
+            component, computed from the qb window functions
+        wbl_cb : dict
+            Window functions for each cb
         """
-        return xft.bin_spec(
```
[the remark this answers is not in the document]
Comment (arahlin) [reply xfaster/xfaster_class.py:4938]: Sure, we can delete it.  I’d left it in for ease of comparison, but you’re right that it’s not used anymore.

**Comment by arahlin** [comment]
> This looks great! It's all consistent with the text in the paper and I think we're good to merge, other than my question about that bin spectrum function-- want to double check you think it's useless now as well!

One thing I haven’t done is update the bandpowers section of the tutorial.  I can probably get around to that today, unless you want to tackle that one!

**Comment by arahlin** [comment]
Ok, I've updated the tutorial with equations for constructing the bandpowers.  Also fixed some issues with how bandpower files on disk are checked and written, so now running the bandpowers in the example script vs in the tutorial notebook produces identical results.

**Pull request merged**